### DISCUSSION TITLE
How to release multi-arch packages?

### DISCUSSION THREAD

**Discussion by radiosilence** [nanovms/ops#1601] [Q&A]
I have a package (radiosilence/nano-web) that I'd like to be able to run on both AWS x86 instances (amd64) and also Gravitron (arm64). Should I release them as separate versions ie `nano-web_x.x.x-amd64` and `nano-web_x.x.x-arm64` or is there a way to have a multi-arch unikernel? Is there an idiomatic way to do this at all?

**Comment by eyberg** [comment]
Yes. We're currently releasing them as separate packages. So when you upload it there is a toggle to choose arm vs x86.

You can see this in action via:

```
 ops pkg search node --arch=amd64
```

```
ops pkg search node --arch=arm64
```

The x86 pkgs clearly outnumber arm by quite a lot.

I suppose one could create a fat package like macos universal/fat binary but the kernel is still purpose built for each architecture and arch has to be chosen at image creation.

**Comment by radiosilence** [comment]
@eyberg Ah yeah, I got stuck at doing `ops pkg push` as it seems like there's no way to set the arch via this CLI and we have that weird issue with x86_64 vs amd64, I did a patch to add the config.json `Arch` field to the push command but it seemed too immature to PR.

I think adding the `--arch` argument to ops pkg push would seem like a sensible solution.

**Reply by eyberg** [reply]
that was added in https://github.com/nanovms/ops/pull/1611

**Reply by radiosilence** [reply]
Ah didn't seem to work quite? I've done a PR but may be dumb and missing something

https://github.com/nanovms/ops/pull/1618

In fact I think the PR you linked I think enabled me to make mine and extend the push command to use it.
